components/new_post/index.tsx

```diff
@@ -0,0 +1,228 @@
+
+    const canReply = isDMorGM || (channel.team_id === currentTeam.id);
+    const directTeammate = getDirectTeammate(state, channel.id);
+
+    const previewCollapsed = get(
+        state,
+        Preferences.CATEGORY_DISPLAY_SETTINGS,
+        Preferences.COLLAPSE_DISPLAY,
+        Preferences.COLLAPSE_DISPLAY_DEFAULT,
+    );
+
+    const previewEnabled = getBool(
+        state,
+        Preferences.CATEGORY_DISPLAY_SETTINGS,
+        Preferences.LINK_PREVIEW_DISPLAY,
+        Preferences.LINK_PREVIEW_DISPLAY_DEFAULT === 'true',
+    );
+
+    return {
+        enableEmojiPicker,
+        enablePostUsernameOverride,
+        isEmbedVisible: isEmbedVisible(state, post.id),
+        isReadOnly: false,
+        teamId,
+        currentUserId: getCurrentUserId(state),
+        isFirstReply: previousPost ? isFirstReply(post, previousPost) : false,
+        hasReplies: getReplyCount(state, post) > 0,
+        replyCount: getReplyCount(state, post),
+        canReply,
+        pluginPostTypes: state.plugins.postTypes,
+        channelIsArchived: isArchivedChannel(channel),
+        isConsecutivePost: isConsecutivePost(state, ownProps),
+        previousPostIsComment,
+        isFlagged: get(state, Preferences.CATEGORY_FLAGGED_POST, post.id, null) != null,
+        compactDisplay: get(state, Preferences.CATEGORY_DISPLAY_SETTINGS, Preferences.MESSAGE_DISPLAY, Preferences.MESSAGE_DISPLAY_DEFAULT) === Preferences.MESSAGE_DISPLAY_COMPACT,
+        colorizeUsernames: get(state, Preferences.CATEGORY_DISPLAY_SETTINGS, Preferences.COLORIZE_USERNAMES, Preferences.COLORIZE_USERNAMES_DEFAULT) === 'true',
+        shouldShowActionsMenu: shouldShowActionsMenu(state, post),
+        showActionsMenuPulsatingDot,
+        shortcutReactToLastPostEmittedFrom,
+        isBot,
+        collapsedThreadsEnabled: isCollapsedThreadsEnabled(state),
+        shouldHighlight: ownProps.shouldHighlight || highlightedPostId === post.id,
+        oneClickReactionsEnabled,
+        recentEmojis: emojis,
+        isCollapsedThreadsEnabled: isCollapsedThreadsEnabled(state),
+        isExpanded: state.views.rhs.isSidebarExpanded,
+        isPostBeingEdited: getIsPostBeingEditedInRHS(state, post.id),
+        isMobileView: getIsMobileView(state),
+        previewCollapsed,
+        previewEnabled,
+        post,
+        channelName: channel.display_name,
+        channelType: channel.type,
+        teamDisplayName,
+        displayName: getDisplayNameByUser(state, directTeammate),
+        teamName,
+    };
+}
+
+function mapDispatchToProps(dispatch: Dispatch<AnyAction>) {
+    return {
+        actions: bindActionCreators({
+            markPostAsUnread,
+            emitShortcutReactToLastPostFrom,
+            setActionsMenuInitialisationState,
+            selectPost,
+            setRhsExpanded,
+            removePost: removePostAndCloseRHS,
+            closeRightHandSide,
+            selectPostCard: selectPostCardFromRightHandSideSearch,
+        }, dispatch),
+    };
+}
```

Comment:
something we discussed in web guild. this can be a plain object

Reply:
sorry I missed this, is there an example in the app that you can show me?

Reply:
components/integrations/delete_integration_link/index.ts